tools/src/tester/StoryOutputs.ts

```diff
@@ -48,12 +48,13 @@ export class StoryOutputs {
   }
 
   resolve_string (str: string): any {
-    const ref = OutputReference.parse(str)
-    if (ref) {
-      return this.get_output_value(ref.chapter_id, ref.output_name)
-    } else {
-      return str
-    }
+    return OutputReference.replace(str, (chapter_id, output_name) => {
```

Comment:
If this is the only place the `replace` function is used, then having a generic replace function that can handle any callback is a bit over-engineered, IMO.

Reply:
I think it's best we keep the parsing regex values/logic in one place so it doesn't leak into the usage of it.